### ISSUE TITLE
How to access all nodes like in d3.selectAll()

### ISSUE THREAD

**Issue by lothorp** [vasturiano/3d-force-graph#42]
I have been trying to access the nodes within the scene, with the idea to pass their [x,y,z] co-ords to my django back end to run a knn model. 

I have it working in a normal d3 force directed graph but im unable to select all nodes in this and apply them to a var.

Thanks in advance!

**Comment by vasturiano** [comment]
Hi @lothorp. You already have those coordinates, if you look in each of the node objects in your `graphData` structure, you'll find each has been annotated with `x`, `y` and `z` attributes representing its current coordinates. You can extract those values to feed to a different system.

Example:
```
{ 
  nodes: [{
    "id": "1",
    "x":-15.31939841608478,
    "y":26.545855541061,
    "z":78.58912678726857,
    ...
    },
    ...
  ],
  links: [...]
}
```

Just a note, this is only true if you're using the `d3` engine. `ngraph` does not function with coordinate object annotation.

**Comment by lothorp** [comment]
This is an example of me attempting to access the graphData in console. 

I get the print out in console however the object is empty.

`<head>
  <style> body { margin: 0; } </style>
  <script src="//unpkg.com/3d-force-graph"></script>
  
</head>

<body>
  <div id="3d-graph"></div>
  <button id="graph-btn">Button</button>
</body>

<script>
    const elem = document.getElementById('3d-graph');
    const Graph = ForceGraph3D()(elem)
      .forceEngine('d3')
      .nodeResolution(8)
      .jsonUrl('pcap_export.json')
      .warmupTicks(10)
      .nodeAutoColorBy('group')
      .nodeLabel( node => `${node.id}`)
      .onNodeHover(node => elem.style.cursor = node ? 'pointer' : null);

    document.getElementById("graph-btn").addEventListener("click", get_node_locs);

    function get_node_locs() {
        var x = Graph.graphData()
        console.log(x)
    }
</script>`

Your repos are great by the way, keep it up!

**Comment by vasturiano** [comment]
@lothorp you're right. This is a bug in the lib, related to how `graphData` is exposed when using the internal fetch in `jsonUrl`. There's nothing wrong with your code. I'll fix the lib when I have just a bit of time.
If you want get unblocked immediately, you can switch back to using the `graphData(<data>)` method to input the data, instead of `jsonUrl()`.

And thanks for the report!

**Comment by vasturiano** [comment]
@lothorp this is now fixed from `v1.28.11. Let me know if you run into any other issues.
